Reduce doc bloat & edit LDAP external authentication user story (#2975)
theforeman/foreman-documentation: guides/common/modules/proc_configuring-project-to-use-ldap.adoc

[id="Configuring_Project_to_Use_LDAP_{context}"]
= Configuring {Project} to use LDAP

Configure an LDAP authentication source to enable users to log in to {Project} with their existing LDAP credentials.

.Prerequisite
* Your user account has a role assigned that has the following permissions assigned:
** `view_authenticators`, `create_authenticators`, `edit_authenticators`
** `view_locations`, `assign_locations`

> ** `view_organizations`, `assign_organizations`

Author: Is this list accurate?

Author: We have different procedures for external groups so we'll cover those permissions there. And yes, the point is to focus the scope only on what's documented. So I think we should be okay here; if it turns out it doesn't work this way, we can always revisit.

Thanks!

[id="Configuring_Project_to_Use_LDAP_{context}"]
= Configuring {Project} to use LDAP

Configure an LDAP authentication source to enable users to log in to {Project} with their existing LDAP credentials.

.Prerequisite
* Your user account has a role assigned that has the following permissions assigned:
** `view_authenticators`, `create_authenticators`, `edit_authenticators`
** `view_locations`, `assign_locations`
** `view_organizations`, `assign_organizations`

.Procedure
. On your {ProjectServer}, set the Network Information System (NIS) service boolean to true to prevent SELinux from stopping outgoing LDAP connections:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# setsebool -P nis_enabled on
----
. In the {ProjectWebUI}, navigate to *Administer* > *Authentication Sources*.
. From the *LDAP* menu, select *Create*.
.. On the *LDAP server* tab, enter the details of your LDAP server.
+

> For TLS encrypted connections, select *LDAPS* to enable encryption.

Reviewer: ```suggestion
Select *LDAPS* for TLS connections.
```
I think the user might skip over this step because it looks like a note.

Reviewer: AFAIK, running LDAP without some kind of encrypted connection is a security risk. However, the wording does sound like TLS is optional. I'm going to assume that SSL is running by default. If that is the case, then your wording is fine.

> Selecting *LDAPS* changes the default port in {ProjectWebUI} to `636`, which is the default for LDAPS.

Reviewer: I don't think you need this sentence. As soon as the user selects *LDAPS*, the port number changes to the correct default. The user does not need to be told that the port is set to the LDAPS default port.

> .. On the *Account* tab, enter the account information and domain name details.

Reviewer: I am not sure whether the user will see the examples in the sections after this procedure. You might want to restore the xrefs.

.. On the *Attribute mappings* tab, map LDAP attributes to {Project} attributes.
.. On the *Locations* tab, select the locations you want {Project} to assign to users created from the LDAP authentication source.
These locations will be available to users after they first log in.
.. On the *Organizations* tab, select the organizations you want {Project} to assign to users created from the LDAP authentication source.

> These locations will be available to users after they first log in.

Author: ```suggestion
These locations are available to users after they log in for the first time.
```

. Click *Submit*.

.Next steps

> * If you did not select *Automatically Create Accounts In {Project}* on the *Account* tab, create user accounts manually.

Author: I considered the mod docs guidelines as I was writing it and concluded it was worth it to bend them a little in this case.

These are still next steps. It's just that the next steps might differ based on what users selected when following the procedure. So from my perspective, it's still just a list of links... only with a little context.

For the record, I also tried including the links from the procedure steps or adding them after the procedure without the `.Next steps` heading. The current version seemed like the best balance between the templates, UX, and minimalism :shrug:

No change, let me know if you disagree with my reasoning.

> * If you did not select *Automatically Create Accounts In {Project}* on the *Account* tab, create user accounts manually.

Reviewer: I think the user understands that "Creating user accounts manually" refers to cases in which user accounts were **not** created automatically. We assume that the user is a trained admin, so it is not necessary to spell out every detail. :-)  That is why I think the "If ..." statement is not necessary.

This is not a blocker for the PR. Just a comment.

----

> # setsebool -P nis_enabled on

Author: Where do I run this command? On the Project Server?

Reviewer 2: Yes, Satellite is the machine that initiates outgoing LDAP query.
